### PR TITLE
redundant block requests

### PR DESCRIPTION
**Pull request by almogdepaz**

### CODE REVIEW

File: sync/worker.go [hunk 1]
```diff
@@ -128,20 +128,21 @@ func NewBlockWorker(s *Syncer, count int, reqFactory BlockRequestFactory, ids ch
 				timeout := time.After(s.RequestTimeout)
 				select {
 				case <-timeout:
-					s.Error("request to %v timed out", peer)
+					s.Error("block %v request to %v timed out", id, peer)
 				case v := <-ch:
-					if v != nil {
+					i, ok := v.(*types.Block)
+					if i != nil || ok {
```
Comment (y0sher) [inline sync/worker.go]: why not && ?

File: sync/worker.go [hunk 2]
```diff
@@ -128,20 +128,21 @@ func NewBlockWorker(s *Syncer, count int, reqFactory BlockRequestFactory, ids ch
...
 						retrived = true
-						s.Info("Peer: %v responded ", peer)
-						s.Debug("Peer: %v response was  %v", v)
+						s.Info("Peer: %v responded to %v block request ", peer, i.ID())
 						output <- v
-						continue
+						goto next
```
Comment (y0sher) [inline sync/worker.go]: `goto`? why not continue?

File: sync/worker_test.go [hunk 3]
```diff
@@ -75,3 +75,42 @@ func TestNewNeighborhoodWorker(t *testing.T) {
...
+	syncObj3.Peers = pm1
+
+	wrk := NewBlockWorker(syncObj3, 1, BlockReqFactory(), blockSliceToChan([]types.BlockID{block.Id}))
+	go wrk.Work()
+	count := 0
+	for {
+		select {
+		case item, ok := <-wrk.output:
+			if !ok {
+				goto finish
```
Comment (y0sher) [inline sync/worker_test.go]: I know its a test. but let's try not using goto if we can. you should give the for loop a label and then break with the label.

File: sync/worker.go (hunk 2 above)
Comment (y0sher) [reply sync/worker.go]: though this is a valid usage for goto. we can break over labels in go so we don't really need to use it.

**Review by y0sher — CHANGES_REQUESTED** [review]
see comments

File: sync/worker.go (hunk 2 above)
Comment (almogdepaz) [reply sync/worker.go]: continue was the whole reason for the bug, as for the break loop i really dont care

File: sync/worker_test.go (hunk 3 above)
Comment (almogdepaz) [reply sync/worker_test.go]: same

**Review by y0sher — APPROVED** [review]

**Pull request merged**